tests/jballerina-unit-test/src/test/resources/test-src/expressions/object/object_constructor_expression.bal

```diff
@@ -125,6 +125,51 @@ function testObjectConstructorIncludedMethod() {
     assertTrue(objWithIncludedMethod.n == 100);
 }
 
+// test distinct id support
+
+distinct class DistinctFoo {
+    int i = 0;
+}
+
+distinct class DistinctFooA {
+    int i = 0;
+}
+
+distinct class DistinctFooB {
+    int i = 0;
+}
+
+function testObjectConstructorWithDistintExpectedType() {
+    DistinctFoo distinctObject = object {
+                                        int i;
+                                        function init() {
+                                            self.i = 20;
+                                        }
+                                    };
+    assertValueEquality(20, distinctObject.i);
+}
+
+function testObjectConstructorWithDistintTypeReference() {
+    DistinctFooA distinctObject = object DistinctFooA {
+                                            int i;
+                                            function init() {
+                                                self.i = 30;
+                                            }
+                                         };
+    assertValueEquality(30, distinctObject.i);
+}
+
+function testObjectConstructorWithDistintTypeReferenceVar() {
+    var distinctObject = object DistinctFooB {
+                                            int i;
+                                            function init() {
+                                                self.i = 25;
+                                            }
+                                         };
+    DistinctFooB newDistinctRef = distinctObject;
+    assertValueEquality(25, newDistinctRef.i);
+}
+
```

Comment:
```ballerina

class K {
    int i = 0;
    int j = 0;
}

class J {
    int i = 0;
}

DistinctFoo | DistinctFooA | K | J distinctObject = object {
                                        int i;
                                        function init() {
                                            self.i = 20;
                                        }
                                    };
```
The above should not be allowed right, as the union is not definite.

Reply:
Yes. Need to fix it.

Reply:
Fixed by 529f4a2f206330e309add582d1d4da2f283dc042